src/rust/engine/fs/store/src/lib.rs

```diff
@@ -104,27 +104,30 @@ pub struct DirectoryMaterializeMetadata {
 }
 
 impl DirectoryMaterializeMetadata {
-  fn to_path_list(&self) -> Vec<String> {
+  fn to_relative_paths(&self) -> Result<HashSet<RelativePath>, String> {
```

Comment:
Thius is the bugfix. The prior `String`s were un-normalized in practice; ie we had "./script.sh" instead of "script.sh", and those weren't comparing as desired below in `contains_file` against the input `RelativePath`.